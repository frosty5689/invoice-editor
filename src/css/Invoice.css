.Invoice {
  max-width: 800px;
  margin: auto;
  padding: 30px;
  border: 1px solid #eee;
  box-shadow: 0 0 10px rgba(0, 0, 0, .15);
  font-size: 16px;
  line-height: 24px;
  font-family: 'Helvetica Neue', 'Helvetica', Helvetica, Arial, sans-serif;
  color: #555;
}
.Invoice table {
  width: 100%;
  line-height: inherit;
  text-align: left;
}

.Invoice td {
  padding: 5px;
  vertical-align: top;
}

.Invoice table tr td:nth-child(5) {
  width: 20px;
}

.Invoice table tr.Invoice-head td {
  background: #eee;
  border-bottom: 1px solid #ddd;
  font-weight: bold;
}


.Invoice table tr.Invoice-item td{
  border-bottom: 1px solid #eee;
}

.Invoice-summary-first td:nth-child(4) {
  border-top: 2px solid #ddd;
}

.Invoice table tr:last-child td:nth-child(4){
  border-bottom: 1px solid #ddd;
}

.Invoice table tr:last-child td:nth-child(5){
  border-bottom: 1px solid #ddd;
}

.Invoice-summary {
  display: flex;
  justify-content: flex-end;
}

.Invoice-summary-detail {
  display: flex;
  justify-content: space-between;
}

.Invoice-summary-detail-total {
  font-weight: bold;
}