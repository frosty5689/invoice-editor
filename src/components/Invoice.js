import React from 'react';

import InvoiceItem from './InvoiceItem';
import InvoiceSummary from './InvoiceSummary';

import '../css/Invoice.css';

class Invoice extends React.Component {
  constructor(props) {
    super(props);

    const items = [{
      name: 'Widget',
      quantity: 2,
      price: 10.00,
    }, {
      name: 'Cog',
      quantity: 3,
      price: 21.00,
    }];

    this.state = {
      items: items,
    }
  }

  handleItemChange(event, index) {
    const target = event.target;
    const name = target.name;
    const items = this.state.items.slice();
    let item = items[index];
    const newValue = target.type === "number" ? Number(target.value) : target.value;
    item[name] = newValue;

    this.setState({
      items: items
    });
  }

  handleItemDelete(index) {
    const items = this.state.items.slice();
    items.splice(index, 1);

    this.setState({
      items: items
    });
  }

  handleItemAdd() {
    const items = this.state.items.slice().concat([{name: '', quantity: 1, price: 0.00}]);
    this.setState({
      items: items
    });
  }

  render() {
    const items = this.state.items;
    const lineItems = items.map((item, index) => {
      return (
        <InvoiceItem key={index} item={item} onItemChange={(e) => this.handleItemChange(e, index)} onItemDelete={() => this.handleItemDelete(index)} />
      );
    });

    return (
      <div className="Invoice">
        <table>
          <thead>
            <tr className="Invoice-head">
              <td>Item</td>
              <td>Qty</td>
              <td>Price</td>
              <td>Total</td>
              <td></td>
            </tr>
          </thead>
          <tbody>
            {lineItems}
            <tr>
              <td><button onClick={() => this.handleItemAdd()}>New Item</button></td>
              <td></td>
              <td></td>
              <td></td>
              <td></td>
            </tr>
          </tbody>
        </table>

        <InvoiceSummary items={items} />
      </div>
    );
  }
}

export default Invoice;