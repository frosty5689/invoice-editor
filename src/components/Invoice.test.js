import React from 'react';
import Invoice from './Invoice';
import renderer from 'react-test-renderer';

jest.mock('./InvoiceItemList', () => {
  const InvoiceItemList = () => <div />;
  return InvoiceItemList;
});

jest.mock('./InvoiceSummary', () => {
  const InvoiceSummary = () => <div />;
  return InvoiceSummary;
});

describe('Invoice component', () => {

  it('Invoice should render correctly', () => {
    const component = renderer.create(
      <Invoice />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});