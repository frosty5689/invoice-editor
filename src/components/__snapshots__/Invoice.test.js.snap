// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice component Invoice should render correctly 1`] = `
<div
  className="Invoice"
>
  <table>
    <thead>
      <tr
        className="Invoice-head"
      >
        <td>
          Item
        </td>
        <td>
          Qty
        </td>
        <td>
          Price
        </td>
        <td>
          Total
        </td>
        <td />
      </tr>
    </thead>
    <div />
  </table>
  <div />
</div>
`;
